managedplugin/metrics.go

```diff
@@ -1,10 +1,57 @@
 package managedplugin
 
-import "sync/atomic"
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sync/atomic"
+)
+
+type DownloadMode int
+
+const (
+	DownloadModeUnknown DownloadMode = iota
+	DownloadModeCached
+	DownloadModeRemote
+)
+
+func (r DownloadMode) String() string {
+	return [...]string{"unknown", "cached", "remote"}[r]
+}
+
+func (r DownloadMode) MarshalJSON() ([]byte, error) {
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(r.String())
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
+
+func (r *DownloadMode) UnmarshalJSON(data []byte) (err error) {
+	var mode string
+	if err := json.Unmarshal(data, &mode); err != nil {
+		return err
+	}
+	if *r, err = DownloadModeFromString(mode); err != nil {
+		return err
+	}
+	return nil
+}
+
+func DownloadModeFromString(s string) (DownloadMode, error) {
+	switch s {
+	case "cached":
+		return DownloadModeCached, nil
+	case "remote":
+		return DownloadModeRemote, nil
+	default:
+		return DownloadModeUnknown, fmt.Errorf("unknown registry %s", s)
```

Comment:
```suggestion
		return DownloadModeUnknown, fmt.Errorf("unknown mode %s", s)
```

Reply:
I changed this string. Although it wasn't something from this PR, but should be fine.